apps/passport-client/components/shared/PCDArgs.tsx

```diff
@@ -56,20 +58,66 @@ export function PCDArgs<T extends PCDPackage>({
   options?: ArgsDisplayOptions<ArgsOf<T>>;
 }): JSX.Element {
   const [showAll, setShowAll] = useState(false);
+
+  // Flatten record arguments (if any), keeping track of the parent argument to
+  // properly mutate (cf. `setArg`) as well as inheriting argument fields from
+  // the record argument.  Validator parameters are also combined.
+  type flattenedArgTriple = [
+    string | undefined,
+    string,
+    Argument<PrimitiveArgumentTypeName>
+  ];
```

Comment:
generally prefer to declare types outside of functions (i.e. in module scope), and for their names to start with capital letters.